qt-models/filtermodels.cpp

```diff
@@ -32,30 +32,38 @@ FilterModelBase::FilterModelBase(QObject *parent) : QStringListModel(parent),
 {
 }
 
-// Update the stringList and the checkState array.
+// Update the stringList and the items array.
 // The last item is supposed to be the "Show Empty Tags" entry.
 void FilterModelBase::updateList(const QStringList &newList)
 {
-	// Keep copy of old checkState array to reimport them later.
-	std::vector<char> oldCheckState = checkState;
-	checkState.resize(newList.count());
-	std::fill(checkState.begin(), checkState.end(), false);
+	// Keep copy of the old items array to reimport the checked state later.
+	// Note that by using std::move(), this is an essentially free operation:
+	// The data is moved from the old array to the new one and the old array
+	// is reset.
+	std::vector<Item> oldItems = std::move(items);
+	items.resize(newList.count());
```

Comment:
So this moves the existing `items` to `oldItems` and then resizes `items` to the new length of the `newList`... but then what's the content of `items`? Are all members zeroed out? Or undefined?
In the loop below you just set `.checked` to false, so I assume the members are undefined?

Reply:
The assignment with `std::move` essentially resets the source-vector to zero size (in this case we could also simply have swapped the new and the old vectors - that would have been the pre-C++11 idiomatic thing to do).

The resize leaves the contents of the struct in an undefined state because we didn't specify a default constructor and the PODs [plain old data type] such as `bool` and `int` don't have a default constructor either. Resize is conceptually the same as `malloc`/`realloc`ing a dynamic array.

I extended the comments.